### DISCUSSION TITLE
What if we added a filter to `HXWP_Render::get_theme_path()`?

### DISCUSSION THREAD

**Discussion by mwender** [TCattd/HTMX-API-WP#5] [Ideas]
We've had an instance where we've included the HTMX-API-WP, and we've had to create a new child theme in order to add our templates to `/htmx-templates/`. Whereas our project already had a plugin we're coding, and that plugin would be a sensible place to store our HTMX templates. It seems like it might be nice to be able to filter the path returned by `HXWP_Render::get_theme_path()`. Something like this:

```
	protected function get_theme_path()
	{
		$theme_path = trailingslashit(get_template_directory());

		if (is_child_theme()) {
			$theme_path = trailingslashit(get_stylesheet_directory());
		}

                apply_filters( 'htmx_template_path', $theme_path );

		return $theme_path;
	}
```

With the above, I believe I could then run the following inside my plugin so that I could store my templates inside `/myplugin/htmx-templates/`:

```
add_filter( 'htmx_template_path', function( $theme_path ){
  return plugin_dir_path( __FILE__ ) . 'htmx-templates/';
});
```

Thoughts?

**Comment by EstebanForge** [comment]
Good idea :)

I think that should be at get_template_file() method.

```
	protected function get_template_file($template_name = '')
	{
		if (empty($template_name)) {
			return false;
		}

		// Let users filter the theme path, so they can place HTMX templates in a different location
		$htmx_templates_path = apply_filters('hxwp/get_template_file/htmx_templates_path', $this->get_theme_path());

		// Add full path and extension to the template name
		$template_path = $htmx_templates_path . HXWP_TEMPLATE_DIR . '/' . $template_name . HXWP_EXT;

		// Sanitize full path
		$template_path = $this->sanitize_full_path($template_path);

		return apply_filters('hxwp/get_template_file/return', $template_path, $template_name);
	}
```

Or is the return filter too much? Don't want 3rd parties to mess with it, really. We could be opening a hole in there in terms of security.

**Comment by EstebanForge** [comment]
Check this out, please: https://github.com/TCattd/HTMX-API-WP/commit/f1c8ac6063b6196ea8efe42085cef6aae75c65ca

I think that should work in your case. Let me know, please.
